src/components/CheckboxButton/index.js

```diff
@@ -0,0 +1,87 @@
+import React, {Component} from 'react';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import CONST from '../../CONST';
+
+const propTypes = {
+    /** Should the input be disabled  */
+    disabled: PropTypes.bool,
+
+    /** A function that is called when the box/label is pressed */
+    onPress: PropTypes.func.isRequired,
+
+    /** Additional styles to add to checkbox button */
+    style: PropTypes.oneOfType([
+        PropTypes.arrayOf(PropTypes.object),
+        PropTypes.object,
+    ]),
+
+    /** A ref to forward to the Pressable */
+    forwardedRef: PropTypes.oneOfType([
+        PropTypes.func,
+        PropTypes.shape({current: PropTypes.instanceOf(React.Component)}),
+    ]),
+
+    /** Children to wrap in CheckboxButton. */
+    children: PropTypes.node.isRequired,
+};
+
+const defaultProps = {
+    disabled: false,
+    style: [],
+    forwardedRef: undefined,
+};
+
+class CheckboxButton extends Component {
+    constructor(props) {
+        super(props);
+
+        this.onClick = this.onClick.bind(this);
+        this.onKeyDown = this.onKeyDown.bind(this);
+        this.onToggle = this.onToggle.bind(this);
+    }
+
+    onClick() {
+        this.onToggle();
+    }
+
+    onKeyDown(event) {
+        if (event.keyCode !== CONST.KEYCODE.SPACE) {
+            return;
+        }
+
+        this.onToggle();
+        event.stopPropagation();
+        event.preventDefault();
+    }
+
+    onToggle() {
+        if (this.props.disabled) {
+            return;
+        }
+
+        this.props.onPress();
+    }
+
+    render() {
+        return (
+            <View
+                focusable
+                onClick={this.onClick}
+                onKeyDown={this.onKeyDown}
+                ref={this.props.forwardedRef}
+                style={this.props.style}
+            >
+                {this.props.children}
+            </View>
```

Comment:
What is the advantage of using View instead of pressable? Could you please summarize the solution in this PR?

Reply:
Sorry, I didn't understand it. 

`It means when it is focused and enter is pressed it will trigger onClick func` then you said `we do not want this behavior instead it should submit the form.` 

Both contradict each other.

Reply:
A button is triggered with `Space, Enter and Click` and Checkbox is toggled with `Space and Click`.

Checkbox does not trigger when enter is pressed but a button does.

Reply:
@mdneyazahmad Can you answer Rajat's questions?

Reply:
Yes, we can do that but I think it is better to use a button as a button and if we need some other element we can build as I did. If you take the source of Pressable it kind of does the same thing.

https://github.com/necolas/react-native-web/blob/dc9ecfbc84c7e6cbf0c36c537ac1b0a86b7e43c1/packages/react-native-web/src/exports/Pressable/index.js#L200

Reply:
We should use the exported API as much as possible. If this possible with standard Pressable than we should use that over anything. Also, doing that will not require two different files for different platforms. 

If this is working. lets do this.

Reply:
@parasharrajat Pressable does not work with `Space`. It only calls on `Enter` and `Click`.  So, checking event is not enough here.

Reply:
You are using the keydown listener to listen for space. Why can't we do this with `Pressable`?

Reply:
We then also have to handle the enter case because we do not want to call onPress. We have to check if onPress is called by enter or click and not call onPress when enter.

Reply:
Ok, that is possible. What do you think @thienlnam? Should we move to `Pressable` or use a View and use a separate View for Non-native as done on the code?

Reply:
I agree we should be using the exported API as much as possible.  We're already using a keydown listener for space and so we should just be able to do that to handle the enter case as well.

Reply:
@thienlnam We do not want to handle enter case here. There should be just two handler space and click.

Reply:
> We then also have to handle the enter case because we do not want to call onPress. We have to check if onPress is called by enter or click and not call onPress when enter.

You said this. So we want to say that we can intercept `Enter` key event as well.

Reply:
That's true if we go with Pressable. We have to intercept and cancel if onPress is triggered with enter. Should I change it to Pressable?

Reply:
Yes, please do it. Let me know if there are other challenges before making the changes. 
Main reason to do this that it would be easier to manage in future.

Reply:
@parasharrajat Let me know if it is fine.
```javascript
function CheckboxButton(props) {
    function onKeyDown(event) {
        if (event.keyCode !== CONST.KEYCODE.SPACE) {
            return;
        }

        props.onPress();
    }

    function checkboxClicked(event) {
        if (event instanceof KeyboardEvent) {
            return;
        }

        props.onPress();
    }

    return (
        <Pressable
            disabled={props.disabled}
            onPress={checkboxClicked}
            ref={props.forwardedRef}
            style={props.style}
            onKeyDown={onKeyDown}
        >
            {props.children}
        </Pressable>
    );
}
```

Reply:
if (event && event.type === 'keydown' && event.key === 'Enter') {